### PR TITLE
Fix dateReadBack test

### LINKED ISSUE #3378: adaptive-expressions dateReadback test fails intermittently
```cmd
 1) expression parser functional test
       Date and time functions
         dateReadBack(timestamp, addDays(timestamp, 1)):

      AssertionError [ERR_ASSERTION]: Acutal and expected don't match. actual is: this Friday, expected is tomorrow, with expression dateReadBack(timestamp, addDays(timestamp, 1))
      + expected - actual

      -this Friday
      +tomorrow

      at assertObjectEquals (libraries\adaptive-expressions\tests\expressionParser.test.js:1189:16)
      at Context.<anonymous> (libraries\adaptive-expressions\tests\expressionParser.test.js:1008:5)
      at processImmediate (internal/timers.js:461:21)
      at process.callbackTrampoline (internal/async_hooks.js:131:14)

  2) expression parser functional test
       Date and time functions
         dateReadBack(addDays(timestamp, 1),timestamp):

      AssertionError [ERR_ASSERTION]: Acutal and expected don't match. actual is: today, expected is yesterday, with expression dateReadBack(addDays(timestamp, 1),timestamp)
      + expected - actual

      -today
      +yesterday

      at assertObjectEquals (libraries\adaptive-expressions\tests\expressionParser.test.js:1189:16)
      at Context.<anonymous> (libraries\adaptive-expressions\tests\expressionParser.test.js:1008:5)
      at processImmediate (internal/timers.js:461:21)
      at process.callbackTrampoline (internal/async_hooks.js:131:14)
```

I haven't looked into this much and maybe it is somehow just my local machine, but these two tests using `dateReadBack` have failed for me the last 5 times I've run them, spread out by about half an hour. I'm calling this "intermittent" because although it failed reliably for me, it doesn't seem to be the case for our pipelines. This is on a fresh clone of the JS repo.

### PR DESCRIPTION
**Pull request by mdrichardson**
Fixes #3378

## Description

`dateReadBack` test was failing when run locally. Turns out, there was a bug in the code that only expressed itself when the machine running the code wasn't on UTC (or something like that).

## What WAS Happening

The issue is how `timestamp1` was getting set.

The code: `const timestamp1 = new Date(dayjs(args[0]).format(dateFormat));`

`args[0]` was a timestamp, `'2018-03-15T13:00:00.111Z'`

1. `dayjs()` call would convert the timestamp to a dayjs Date object (good so far)
2. It would then get formatted to `2018-03-15` (good so far)
3. But then, we'd call `new Date()` on `2018-03-15`, which interprets the input as UTC and creates a Date object in PST, so it would more or less become `2018-03-14` and be off a whole day

## Testing

It now passes, locally:

image

### CODE REVIEW

File: libraries/adaptive-expressions/src/builtinFunctions/dateReadBack.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class DateReadBack extends ExpressionEvaluator {
             const dateFormat = 'YYYY-MM-DD';
             let error = InternalFunctionUtils.verifyISOTimestamp(args[0]);
             if (!error) {
-                const timestamp1 = new Date(dayjs(args[0]).format(dateFormat));
+                const timestamp1 = dayjs(args[0]).toDate();
```
Comment (joshgummersall) [inline libraries/adaptive-expressions/src/builtinFunctions/dateReadBack.ts:37]: @Danieladu can you make sure this makes sense? I don't have all the context on this package.

**Review by Danieladu — DISMISSED** [review]

**Comment by Danieladu** [comment]
Thanks @mdrichardson . Date constructor with string is indeed exists the issue of  parsing with different standard time. So in the [JS doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date), is is not suggested.
> Note: Parsing of date strings with the Date constructor (and Date.parse, they are equivalent) is strongly discouraged due to browser differences and inconsistencies.

We would re-visit all the Data constructor with string parameter, and make sure they are safe in different scenarios.

**Comment by cosmicshuai** [comment]
I run the test on a US west VM for several times. And did not see the test failed.
OS: Ubuntu 18.04
Node version: 14.16.0
@mdrichardson  What is your test platform and the version of node?

**Comment by stevengum** [comment]
image

@cosmicshuai I've run the test at 4:20pm UTC-8, Node.js version is 14.15.0 on Windows build 10.0.19042

**Comment by joshgummersall** [comment]
@cosmicshuai most VMs will be set to UTC rather than a local timezone. I see test failures on my development machine which runs Mac OSX and Node 14 LTS.

**Comment by cosmicshuai** [comment]
@stevengum  @joshgummersall 
I see, you are right. I change the timezone on my local machine manually. The error does exist.

**Review by Danieladu — APPROVED** [review]

**Pull request merged**